### ISSUE TITLE
Update the benchmarks in a nightly build only

### ISSUE THREAD

**Issue by ldionne** [ldionne/metabench#65]
Right now, the full benchmarks get generated each time we push to master. This is inconvenient, because the CI takes a long time to complete. It is also useless, because we already have testing that should catch errors.

What we'd really like is to only run the benchmark jobs (and update gh-pages with the data) once per day. In theory, we could use a service like [nighli.es](http://nightli.es) to achieve this, but I don't think it is possible to know from the CI script that we're being launched by the service. Hence, we wouldn't know when to actually run the benchmarks.

**Comment by ldionne** [comment]
I tried using Nightli.es, but there is no way to know that we're being run as a nightly build. So I've set up a basic Heroku app which performs the following command every night:

```bash
curl -s -X POST 
  -H "Content-Type: application/json" 
  -H "Accept: application/json" 
  -H "Travis-API-Version: 3" 
  -H "Authorization: token ${TRAVIS_TOKEN}" 
  -d '{"request": {
    "message": "Automated nightly update of the benchmarks.", 
    "branch":"master",                                                      
    "config": { "before_install": "export METABENCH_NIGHTLY_BUILD=true" }   
  }}'                                                                         
  https://api.travis-ci.org/repo/ldionne%2Fmetabench/requests
```

This uses the Travis CI API, as documented [here](https://docs.travis-ci.com/user/triggering-builds), to start a build. I tried several different ways to make the script aware that it's being run in a nightly build, but the only thing that worked was to set `METABENCH_NIGHTLY_BUILD=true` explicitly. It's hacky, but it works and I documented the hack properly so it does not bite us in the future. If nightli.es gets the ability to set an environment variable, we'll be able to easily change to it and get rid of the Heroku app.

So the benchmarks should start getting updated tonight. The build will fail for Clang 3.8 because of https://github.com/travis-ci/apt-source-whitelist/pull/255, but everything should work just fine otherwise.
